sympy/ntheory/tests/test_residue.py

```diff
@@ -338,6 +339,7 @@ def test_binomial_p_pow():
         assert _binomial_mod_prime_power(n, m, prime, power) == binomials[m] % prime**power
 
 
+@pytest.mark.thread_unsafe(reason="warning filter broken under multithreading")
```

Comment:
Over in NumPy we just deprecated our warning control utilities in favor of pytest and the standard library: https://github.com/numpy/numpy/pull/29550

Both are thread-safe in Python 3.14 on the free-threaded build and you can opt-in on the GIL-enabled build via a command-line flag: https://docs.python.org/3/whatsnew/3.14.html#concurrent-safe-warnings-control

You can also configure pytest-run-parallel to automatically mark functions using `warns_deprecated_sympy` as thread-unsafe: https://github.com/Quansight-Labs/pytest-run-parallel?tab=readme-ov-file#caveats

Reply:
This is a hangover from sympy's old test runner but it does just use `catch_warnings` so it should work under the same conditions. For now perhaps it is preferable to use
```
pytest -X context_aware_warnings --parallel-threads=4 ...
```
Or perhaps it can be added in pytest.conf:
https://github.com/sympy/sympy/blob/1d7fcba31c84aff49e5b2df7b58119efe41431c4/pyproject.toml#L1-L3